### ISSUE TITLE
Naming/MemoizedInstanceVariableName false positive with multiple variables

### ISSUE THREAD

**Issue by thegorgon** [rubocop-hq/rubocop#6271]
With a method that instantiates more than 1 instance variable using `||=`, the second variable gets triggered by `Naming/MemoizedInstanceVariableName` despite clearly not being a memoization. 

The presence of 2 variable instantiations should be sufficient to disqualify this pattern from the Cop.

## Expected behavior

Expect the following pattern to not trigger a failure of `Naming/MemoizedInstanceVariableName`:

```
def method_name
  @variable1 ||= {}
  @variable2 ||= {}
end
```

## Actual behavior

```
def method_name
  @variable1 ||= {}
  @variable2 ||= {} # This line fails the Cop.
end
```

## Steps to reproduce the problem

Define a method that instantiates two variables using `||=`.

## RuboCop version

```
bundle exec rubocop -V
=> 0.52.1 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-darwin17)
```

**Comment by Eneroth3** [comment]
I'm quite new to the memoization pattern, but to me that looks as if it could have been intended as memoization. I had a similar situation that I solved by setting an explicit return value.
